From 19efb3c7e13c1d1d94d4c3b0f63c8cfd2f4fe66e Mon Sep 17 00:00:00 2001
From: "Ross-Ashikyan, William" <washikyan@maxlinear.com>
Date: Mon, 30 Sep 2024 10:25:19 -0700
Subject: [PATCH] WLANRTSYS-84784 Changed send_afc_request behavior to first
 clean prior AFC data

Issue: AFC send request command would only work first time after 6GHz interface comes up

RC: Open-source default behavior for AFC is to not request new AFC report before timeout of
last report.

Fix: Add a call to clear the prior AFC report upon request for new report.

FIXUP to WLANRTSYS-83162 Ported afcd and composite mode changes from legacy hostapd
---
 hostapd/mxl_ctrl_iface.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hostapd/mxl_ctrl_iface.c b/hostapd/mxl_ctrl_iface.c
index c0c0b1fd7..522875ce8 100644
--- a/hostapd/mxl_ctrl_iface.c
+++ b/hostapd/mxl_ctrl_iface.c
@@ -84,6 +84,7 @@ struct hostapd_data *mxl_get_bss_index(const char **cmd, struct hostapd_data *ha
 static int mxl_hostapd_ctrl_iface_send_afcd_request(struct hostapd_iface *iface,
 					     char *buf, size_t buflen)
 {
+	mxl_hostapd_afc_delete_data_from_server(iface);
 	return mxl_hostapd_afc_send_receive(iface);
 }
 #endif /* CONFIG_AFC */
-- 
2.43.0

